src/eon_type_error.erl

```diff
@@ -50,6 +50,7 @@ binary_join(List, Sep) ->
 
 %%%_* Tests ============================================================
 -ifdef(TEST).
+-ifdef(kivra_nova).
```

Comment:
all of the tests below rely on `kivra_nova` modules like `type_string` etc.

Reply:
It's possible to check for a `define` if some `kivra_nova` were to define one. Under the condition it's compiled before `eon`. Which in itself would be the non-intuitive order (if possible). So, that's not the solution you're asking for...

I wrote this module in ´kivra_core` but then it was moved here...

Reply:
I added a conditional now so those tests only run if `kivra_nova` is in the code path. Would that be a practical solution for now? @zigge-zagge

Reply:
I think that looking at the codepath is an acceptable solution. I am not sure if there any better and more straightforward ways of determining if a library is included in a project? 🤔

Reply:
Is it possible to use kivra_nova's types without using eon? If not, maybe kivra_nova should be absorbed into the eon library? 🤔

Reply:
I think it's good to have them as separate libraries since `eon` is useful without `kivra_nova` (also `eon` is open source, `kivra_nova` isn't). From what I understand, `kivra_nova` isn't very useful by itself, but I think that's okay as long as `eon`'s API is well defined. the only thing I find odd is that `kivra_nova` types are used to run the tests in `eon`. maybe we could just move those tests in `eon`'s `test` directory, together with some mock types that conform with `eon`'s API. @jakobsvenning @zigge-zagge

Reply:
> I think it's good to have them as separate libraries since `eon` is useful without `kivra_nova` (also `eon` is open source, `kivra_nova` isn't). From what I understand, `kivra_nova` isn't very useful by itself, but I think that's okay as long as `eon`'s API is well defined. the only thing I find odd is that `kivra_nova` types are used to run the tests in `eon`. maybe we could just use the test types in `eon`'s `test` directory instead. @jakobsvenning @zigge-zagge

Right, I did not think about the other way around. Yes merging is a bad idea.

I was not aware of that there existed a couple of test types in the test directory in eon, it sounds like a good idea to use these instead of types defined in kivra_nova if possible 👍

Reply:
I added three simple types `test_string`, `test_strings` and `test_address` and changed the tests to use those types instead. I think that makes much more sense than checking for `kivra_nova` 👍